### PR TITLE
Misc: Add helper functions to genesis.

### PR DESCRIPTION
**Pull request by winder**
## Summary

While working on Indexer we found some of the genesis operations to require duplicate or awkward code. Adding these methods to the genesis object simplifies things.

## Test Plan

New unit test and existing regression tests.

### CODE REVIEW

File: data/bookkeeping/genesis.go [hunk 1]
```diff
@@ -101,6 +101,51 @@ func (genesis Genesis) ID() string {
...
+	rewardsPool, err := basics.UnmarshalChecksumAddress(genesis.RewardsPool)
+	if err != nil {
+		return GenesisBalances{}, fmt.Errorf("cannot parse rewards pool addr %s: %w", genesis.RewardsPool, err)
+	}
+
+	return MakeTimestampedGenesisBalances(genalloc, feeSink, rewardsPool, genesis.Timestamp), nil
+}
+
+// Block computes the genesis block.
+func (genesis Genesis) Block() (Block, error) {
```
Comment (algorandskiy) [inline data/bookkeeping/genesis.go:140]: not used

**Review by algorandskiy — COMMENTED** [review]
LGTM

File: data/bookkeeping/genesis.go (hunk 1 above)
Comment (winder) [reply data/bookkeeping/genesis.go:140]: We'd like to call this one from Indexer.

Comment (algorandskiy) [reply data/bookkeeping/genesis.go:140]: call once on initialization? If calling often then the `Balances()` need to memoize its result

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**